### PR TITLE
User management: Add revoke session, verify session, and revoke all sessions

### PR DESCRIPTION
**Pull request by AshGodfrey**
## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: pkg/users/client.go [hunk 1]
```diff
@@ -392,3 +392,132 @@ func (c *Client) CreateUser(ctx context.Context, opts CreateUserOpts) (User, err
...
+		strings.NewReader(encodedForm.Encode()),
+	)
+	if err != nil {
+		return VerifySessionResponse{}, err
+	}
+
+	// Add headers and context to the request
+	req = req.WithContext(ctx)
+	req.Header.Set("User-Agent", "workos-go/"+workos.Version)
+	req.Header.Set("Authorization", "Bearer "+c.APIKey)
```
Comment (jonatascastro12) [inline pkg/users/client.go:423]: This one doesn't require `Authorization` too 😄

File: pkg/users/client.go [hunk 2]
```diff
@@ -392,3 +392,132 @@ func (c *Client) CreateUser(ctx context.Context, opts CreateUserOpts) (User, err
...
+	var body VerifySessionResponse
+	dec := json.NewDecoder(res.Body)
+	err = dec.Decode(&body)
+
+	return body, err
+}
+
+type RevokeSessionOpts struct {
+	SessionToken string `json:"session_token"`
+	SessionID    string `json:"session_id"`
```
Comment (jonatascastro12) [inline pkg/users/client.go:447]:
```suggestion
	SessionToken string `json:"session_token,omitempty"`
	SessionID    string `json:"session_id,omitempty"`
```

File: pkg/users/client.go (hunk 1 above)
Comment (AshGodfrey) [reply pkg/users/client.go:423]: Do users not need their API key for this one then?

Comment (jonatascastro12) [reply pkg/users/client.go:423]: Yes, they don't.

**Review by jonatascastro12 — APPROVED** [review]

**Pull request merged**